pkg/cfgfile/cfgfile_small_test.go

```diff
@@ -0,0 +1,207 @@
+// +build small
+
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2016 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cfgfile
+
+import (
+	"encoding/json"
+	"errors"
+	"testing"
+
+	"github.com/ghodss/yaml"
+	"github.com/intelsdi-x/snap/core/serror"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+const (
+	MOCK_CONSTRAINTS = `{
+		"$schema": "http://json-schema.org/draft-04/schema#",
+		"title": "test schema",
+		"type": ["object", "null"],
+		"properties": {
+			"Foo": {
+				"type": "string"
+			},
+			"Bar": {
+				"type": "string"
+			}
+		}
+	}
+	`
+	INVALID_MOCK_CONSTRAINTS = `{
+		"$schema": "http://json-schema.org/draft-04/schema#",
+		"title": "test schema",
+		"type": ["object", "null"],
+		"properties": {
+			"Foo": {
+				"type": "boolean"
+			},
+			"Bar": {
+				"type": "string"
+			}
+		}
+	}
+	`
+)
+
+type testConfig struct {
+	Foo string
+	Bar string
+}
+
+// create an enumeration of the response types we expect back from
+// our mockReader
+const (
+	JSON = iota
+	YAML
+	FILE_NOT_FOUND
+	INVALID_YAML
+	UNMATCHED_SCHEMA
+)
+
+// create a type for the entries in our test-table
+type ttEntry struct {
```

Comment:
Happy to change it to `entry` instead of `ttEntry` if you feel that it makes things clearer; had named it with the `tt` prefix to try to link the purpose of that type to the `tt` (test table) variable that is defined a bit lower down in this file. Thoughts?